src/main/java/vazkii/patchouli/common/util/RotationUtil.java

```diff
@@ -44,15 +46,8 @@ public static Rotation rotationFromFacing(Direction facing) {
 		}
 	}
 
-	public static Rotation fixHorizontal(Rotation rot) {
-		switch (rot) {
-		case CLOCKWISE_90:
-			return Rotation.COUNTERCLOCKWISE_90;
-		case COUNTERCLOCKWISE_90:
-			return Rotation.CLOCKWISE_90;
-		default:
-			return rot;
-		}
+	public static BlockState rotateState(BlockState state, Rotation rotation) {
+		return state.rotate(rotation).mirror(Mirror.FRONT_BACK);
```

Comment:
can you explain the fix?

Reply:
this lets the blocks handle the rotation better because they can take the direction into account

Reply:
Would it make sense then to have two method calls of this given that a lot of the calls use `world#getBlockState(BlockPos)` so given you have a world and position shouldn't those calls be rotating via `IForgeBlock#rotate(IWorld, BlockPos, Rotation)` if the goal of this is to give blocks more control over how they are rotated? That way when you have world information you can use it, otherwise you just use the `BockState#rotate(Rotation)` one this PR is changing it to for when you don't have a world and position.

Reply:
I implemented your suggestion